test/distributed/org/apache/cassandra/distributed/api/ICoordinator.java

```diff
@@ -27,7 +27,11 @@
 {
     // a bit hacky, but ConsistencyLevel draws in too many dependent classes, so we cannot have a cross-version
     // method signature that accepts ConsistencyLevel directly.  So we just accept an Enum<?> and cast.
-    Object[][] execute(String query, Enum<?> consistencyLevel, Object... boundValues);
+    default Object[][] execute(String query, Enum<?> consistencyLevel, Object... boundValues)
+    {
+        return executeWithResult(query, consistencyLevel, boundValues).toObjectArrays();
+    }
+    ResultSet executeWithResult(String query, Enum<?> consistencyLevel, Object... boundValues);
```

Comment:
I am a fan of getting rid of it, but Object[][] used in places other than this so might be another large refactor to change everything it touches and the methods like it

Reply:
+1 Please get rid of `Object[][]`.

Reply:
I hate Object[][], but I rather not do in this JIRA.  I don't mind pushing to another JIRA and doing that work after this merges, but I don't want to block this on that.

> In a long run, do we even want to keep object[][]?

as a user Object[][] is unusable since the column order != table definition order, so I have no way of knowing what the 3rd column is!